lib/marked.js

```diff
@@ -554,9 +554,48 @@ inline.normal = merge({}, inline);
 inline.pedantic = merge({}, inline.normal, {
   strong: /^__(?=\S)([\s\S]*?\S)__(?!_)|^\*\*(?=\S)([\s\S]*?\S)\*\*(?!\*)/,
   em: /^_(?=\S)([\s\S]*?\S)_(?!_)|^\*(?=\S)([\s\S]*?\S)\*(?!\*)/,
-  link: edit(/^!?\[(label)\]\(\s*<?([\s\S]*?)>?(?:\s+(['"][\s\S]*?['"]))?\s*\)/)
-    .replace('label', inline._label)
-    .getRegex(),
+  link: {
+    exec: function (s) {
```

Comment:
Might worth adding some doc blocks to introduce the why behind some of this...nothing too major, just to help those new to the code.